### ISSUE TITLE
Test get ignored when running in vs code in combination with [ignore] macro

### ISSUE THREAD

**Issue by jb-alvarado** [palfrey/serial_test#93]
Hello,

I would like to use a test like:

```Rust
#[test]
#[serial]
#[ignore]
fn my_special_test() {...}
```

When I run command line:

```
cargo test -- --ignored
```

everything works like suspected. But when I try to run a single test in vs code with:
![run_rest](https://user-images.githubusercontent.com/2212056/217784883-4f217f3f-ea4c-43e9-abdd-ddd6d04d1268.png)
clicking `Run Test`, this test get ignored.

Without `#[serial]` macro the test run through.

Do you have any idea, how to fix this?

**Comment by palfrey** [comment]
Not sure. I've been poking at this a bit, and trying to figure out what the actual invocation VS Code  is using i.e. what Rust command line it's running. We ignore the `ignore` attributes when doing some of the processing as they get dealt with by the stock test runner, but I'm guessing VS Code does some other magic and I'm not sure what that is, which will be the key to figuring this out.

**Comment by jb-alvarado** [comment]
Yes I have seen now that it looks like, it is an issue with vs code. Because when no `#[serial]` marco is set, it runs the command:

```
cargo test --package tests --test test_file --all-features -- test_function --exact --nocapture --ignored
```

But when is set, vs code runs:

```
cargo test --package tests --test test_file --all-features -- test_function --exact --nocapture
```

**Comment by palfrey** [comment]
This might be due to some stuff serial_test does. Can you test out your case with the code from https://github.com/palfrey/serial_test/pull/94 and see if it fixes things?

**Comment by jb-alvarado** [comment]
Yes I can test this, but how can I use this pull request easily?

**Comment by jb-alvarado** [comment]
I have test now #94 and it seems that it works.

**Comment by palfrey** [comment]
> I have test now #94 and it seems that it works.

Hmm. Interesting. So, I know kinda why that code was there to begin with, but I can't reproduce the problem anymore. Let me prod the original reporters...

**Comment by jb-alvarado** [comment]
Thank you!

**Comment by palfrey** [comment]
Released as 2.0.0
